blocklist-client/src/client/risk_client.rs

```diff
@@ -0,0 +1,352 @@
+    debug!("Address registered: {}", register_response.address);
+
+    // If registration is successful, proceed to check the address
+    let RiskAssessment { severity, reason } = get_risk_assessment(client, config, address).await?;
+    debug!(
+        "Received risk assessment: Severity = {}, Reason = {}",
+        severity, reason
+    );
+
+    let is_severe = matches!(severity, RiskSeverity::Severe);
+    let blocklist_status = BlocklistStatus {
+        // `is_blocklisted` is set to true if risk is Severe
+        is_blocklisted: is_severe,
+        severity: severity.to_string(),
+        // `accept` is set to false if severity is Severe
+        accept: !is_severe,
+        reason,
+    };
+
+    Ok(blocklist_status)
+}
+
+/// Evaluates the HTTP response from an API request and translates HTTP status codes into application-specific errors.
+async fn check_api_response(response: Response) -> Result<Response, Error> {
+    match response.status() {
+        StatusCode::OK | StatusCode::CREATED => Ok(response),
+        StatusCode::BAD_REQUEST => Err(Error::HttpRequestErr(
+            response.status(),
+            "Bad request - Invalid parameters or data".to_string(),
+        )),
+        StatusCode::FORBIDDEN => Err(Error::Unauthorized),
+        StatusCode::NOT_FOUND => Err(Error::NotFound),
+        StatusCode::NOT_ACCEPTABLE => Err(Error::NotAcceptable),
+        StatusCode::CONFLICT => Err(Error::Conflict),
+        StatusCode::INTERNAL_SERVER_ERROR => Err(Error::InternalServerErr),
+        StatusCode::SERVICE_UNAVAILABLE => Err(Error::ServiceUnavailable),
+        StatusCode::REQUEST_TIMEOUT => Err(Error::RequestTimeout),
+        status => Err(Error::HttpRequestErr(
+            status,
+            "Unhandled status code".to_string(),
+        )),
+    }
+}
+
+fn register_address_path(base_url: &str) -> String {
+    format!("{}{}", base_url, API_BASE_PATH)
+}
+
+fn risk_assessment_path(base_url: &str, address: &str) -> String {
+    format!("{}{}/{}", base_url, API_BASE_PATH, address)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::common::RiskSeverity::{Low, Severe};
+    use mockito::{mock, server_url, Mock};
+
+    const TEST_ADDRESS: &str = "test_address";
+    const ADDRESS_REGISTRATION_BODY: &str = r#"{"address": "test_address"}"#;
+
+    // Setup function for common client and configuration
+    fn setup_client() -> (Client, RiskAnalysisConfig) {
+        let client = Client::new();
+        let config = RiskAnalysisConfig {
+            api_url: server_url(),
+            api_key: "dummy_api_key".to_string(),
+        };
+        (client, config)
+    }
+
+    // Helper function to setup a mock API response
+    fn setup_mock(method: &str, path: &str, status: u16, body: &str) -> Mock {
+        return mock(method, path)
+            .with_status(status.into())
+            .with_header("content-type", "application/json")
+            .with_body(body)
+            .create();
+    }
```

Comment:
I'm not super familiar with `mockito` (just looking through their docs now), but I think you'll want to use `.create_async` here.

Reply:
`create_async` was part of newer versions. I tried updating the version and the test logic but I couldn't get the mock to work properly...wrestled with it for some time but keeping it the way it is for now.